### PR TITLE
Fix scenario in which an exception would prevent the channel response…

### PR DESCRIPTION
**Pull request by yachtcaptain23**
… from being added to the channel_responses array.

Closes https://github.com/brave-intl/publishers/issues/3082

### CODE REVIEW

**Review by dlipeles — APPROVED** [review]

File: app/jobs/channels/approve_channel_transfer_job.rb [hunk 1]
```diff
@@ -11,7 +11,7 @@ def perform(channel_id:)
         ## There is a helper function in channel.rb that can help minimize this code
         ## Need to generate new bF deposit ID on channel transfer:
         channel.deposit_id = nil
-        if contested_by.publisher.selected_wallet_provider_type == BITFLYER_CONNECTION
+        if contested_by.publisher.selected_wallet_provider_type == Channel::BITFLYER_CONNECTION
```
Comment (dlipeles) [inline app/jobs/channels/approve_channel_transfer_job.rb:14]: quick q, does Channel::BITFLYER_CONNECTION return 'BitflyerConnection'

File: app/jobs/cache/browser_channels/responses_for_prefix.rb [hunk 2]
```diff
@@ -38,18 +38,17 @@ def generate_brotli_encoded_channel_response(prefix:)
           wallet.paypal_wallet = paypal_wallet
           channel_response.wallets.push(wallet)
         end
-        if site_banner_lookup.publisher.bitflyer_connection.present?
+        if site_banner_lookup.publisher&.bitflyer_connection.present?
```
Comment (dlipeles) [inline app/jobs/cache/browser_channels/responses_for_prefix.rb:41]: I used to have this line here, which might help too 

site_banner_lookup.publisher.selected_wallet_provider_type == BITFLYER_CONNECTION

if we wanted to try that

File: app/jobs/channels/approve_channel_transfer_job.rb (hunk 1 above)
Comment (yachtcaptain23) [reply app/jobs/channels/approve_channel_transfer_job.rb:14]: Yeah it does. 

```
  BITFLYER_CONNECTION = "BitflyerConnection".freeze
```

File: app/jobs/cache/browser_channels/responses_for_prefix.rb (hunk 2 above)
Comment (yachtcaptain23) [reply app/jobs/cache/browser_channels/responses_for_prefix.rb:41]: I'm reverting, this change wasn't necessary

**Pull request merged**